examples/image_decoders/jpeg_viewer.js

```diff
@@ -27,56 +27,40 @@ const jpegCtx = jpegCanvas.getContext("2d");
 
 // Load the image data, and convert it to a Uint8Array.
 //
-let nonBinaryRequest = false;
-const request = new XMLHttpRequest();
-request.open("GET", JPEG_IMAGE, false);
-try {
-  request.responseType = "arraybuffer";
-  nonBinaryRequest = request.responseType !== "arraybuffer";
-} catch (e) {
-  nonBinaryRequest = true;
-}
-if (nonBinaryRequest && request.overrideMimeType) {
-  request.overrideMimeType("text/plain; charset=x-user-defined");
-}
-request.send(null);
-
-let typedArrayImage;
-if (nonBinaryRequest) {
-  const str = request.responseText,
-    length = str.length;
-  const bytes = new Uint8Array(length);
-  for (let i = 0; i < length; ++i) {
-    bytes[i] = str.charCodeAt(i) & 0xff;
+(async function () {
+  const response = await fetch(JPEG_IMAGE);
+  const requestSuccessful = response.ok;
+  if (!requestSuccessful) {
+    throw new Error(`Failed to fetch ${JPEG_IMAGE} - ${response.statusText}`);
   }
-  typedArrayImage = bytes;
-} else {
-  typedArrayImage = new Uint8Array(request.response);
-}
 
-// Parse the image data using `JpegImage`.
-//
-const jpegImage = new pdfjsImageDecoders.JpegImage();
-jpegImage.parse(typedArrayImage);
+  const arrayBuffer = await response.arrayBuffer();
+  const typedArrayImage = new Uint8Array(arrayBuffer);
```

Comment:
Please just make the *exact* changes mentioned in https://github.com/mozilla/pdf.js/pull/14167#discussion_r732466584 here!